container/container.go

```diff
@@ -1,105 +1,241 @@
+// clientHTTPTimeout is the time limit for requests made by docker HTTP client.
+const clientHTTPTimeout = 10 * time.Second
+
+// defaultStopGracePeriod is the timeout value between stopping a container and killing it.
+const defaultStopGracePeriod = 10 * time.Second
+
+var errSwarmNotInit = errors.New(`docker swarm is not initialized. run "docker swarm init" and try again`)
+
+// Status of the service.
+type Status uint
+
+// Possible status for services.
+const (
+	UNKNOWN Status = iota
+	STOPPED
+	STARTING
+	RUNNING
 )
 
+// statuses is a struct used to map service and contaienr statuses.
+var statuses = []struct {
+	container bool
+	service   bool
+	status    Status
+}{
+	{service: true, container: true, status: RUNNING},
+	{service: true, container: false, status: STARTING},
+	{service: false, container: true, status: RUNNING}, // This is actually stopping
+	{service: false, container: false, status: STOPPED},
+}
+
+// buildResponse is the object that is returned by the docker build api.
+type buildResponse struct {
+	Stream string `json:"stream"`
+	Error  string `json:"error"`
+}
+
 // Container describes the API of container package.
 type Container interface {
 	Build(path string) (tag string, err error)
 	CreateNetwork(namespace string) (id string, err error)
 	DeleteNetwork(namespace string) error
-	FindContainer(namespace string) (types.ContainerJSON, error)
-	FindNetwork(namespace string) (types.NetworkResource, error)
-	FindService(namespace string) (swarm.Service, error)
-	ListServices(labels ...string) ([]swarm.Service, error)
-	ListTasks(namespace string) ([]swarm.Task, error)
-	Namespace(namespace string) string
-	SharedNetworkID() (networkID string, err error)
+	SharedNetworkID() string
 	StartService(options ServiceOptions) (serviceID string, err error)
-	Status(namespace string) (StatusType, error)
 	StopService(namespace string) (err error)
-	TasksError(namespace string) ([]string, error)
 	DeleteVolume(name string) error
 }
 
 // DockerContainer provides high level interactions with Docker API for MESG.
 type DockerContainer struct {
 	// client is a Docker client.
-	client docker.CommonAPIClient
-
-	// callTimeout is the timeout value for Docker API calls.
-	callTimeout time.Duration
-
-	// defaultStopGracePeriod is the timeout value between stopping a container and killing it.
-	defaultStopGracePeriod time.Duration
+	client client.CommonAPIClient
 
 	// namespace prefix.
 	nsprefix string
-}
 
-// Option is a configuration func for Container.
-type Option func(*DockerContainer)
+	// sharedNetworkID is cache for an id network.
+	sharedNetworkID string
+}
 
 // New creates a new Container with given options.
-func New(nsprefix string, options ...Option) (*DockerContainer, error) {
+func New(nsprefix string) (*DockerContainer, error) {
+	client, err := client.NewClientWithOpts(client.FromEnv, client.WithTimeout(clientHTTPTimeout))
```

Comment:
Setting up the timeout on the whole docker client doesn't allow any build that take more than `clientHTTPTimeout` (10sec in current implementation).
Do we really need a timeout anyway?
```suggestion
	client, err := client.NewClientWithOpts(client.FromEnv)
```

Reply:
Done